src/Razor/Microsoft.AspNetCore.Mvc.Razor.Extensions/src/ViewComponentTagHelperTargetExtension.cs

```diff
@@ -170,13 +167,33 @@ private void WriteProcessMethodString(CodeWriter writer, TagHelperDescriptor tag
             }
         }
 
-        private string[] GetMethodParameters(TagHelperDescriptor tagHelper)
+        private void WriteProcessInvokeAsyncArgsMethodString(CodeWriter writer, TagHelperDescriptor tagHelper)
+        {
+            var methodReturnType = "Dictionary<string, object>";
+            using (writer.BuildMethodDeclaration(
+                "private",
+                methodReturnType,
+                TagHelperProcessInvokeAsyncArgsMethodName,
+                new Dictionary<string, string>() { { TagHelperContextTypeName, TagHelperContextVariableName } }))
+            {
+                writer.WriteStartAssignment($"{methodReturnType} args").WriteLine("new();");
+                for (var i = 0; i < tagHelper.BoundAttributes.Count; i++)
+                {
+                    var attributeName = tagHelper.BoundAttributes[i].Name;
+                    var parameterName = tagHelper.BoundAttributes[i].GetPropertyName();
+                    writer.WriteLine($"if (__context.AllAttributes.ContainsName(\"{attributeName}\"))");
+                    writer.WriteLine("{");
+                    writer.WriteLine($"args[nameof({parameterName})] = {parameterName};");
```

Comment:
Lets render this in its indented form. Reason being when we run formatting we have to "throw out" anything extra the C# formatter does and if we can avoid that it's a net win for readability + post-processability at tooling time 😄.

I believe code writer has a scope that you can increment and it'll auto-indent for you.

Reply:
> I believe code writer has a scope that you can increment and it'll auto-indent for you.

Thanks! I was having trouble figuring out how to do this. I figured there has to be a way since the method writing methods indent properly but wasn't sure where that was.